community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -322,6 +348,52 @@ def ignore_err(e) -> bool:
     )
 
 
+def sync_mig_groups():
+    lkp = lookup()
+    zone = lkp.zone.split("/")[-1]
+    migs = mig.migs(lkp, zone)
+
+    compute_instances = {
+        name for name, inst in lkp.instances().items() if inst.role == "compute"
+    }
+    slurm_nodes = set(lkp.slurm_nodes().keys())
+
+    migs_to_delete = []
+    for mig_name, mig_obj in migs.items():
+        cluster, nodeset, _ = mig_obj.name.split('-')
+        if cluster != lkp.cfg.slurm_cluster_name:
+            continue
+
+        if not mig.is_stable(lkp, mig_obj):
+            log.info(f"Mig {mig_name} isn't stable yet")
+            continue
+
+        mig_nodes = []
+        for node in slurm_nodes:
+            if (lkp.node_prefix(node) == f"{cluster}-{nodeset}") & mig.is_mig_node(node):
+                mig_nodes.append(node)
+
+        if len(mig_nodes) == 0:
+            # Delete MIG that is attributed to the cluster (by name) but not existing nodeset (result of nodeset deletion)
```

Comment:
Let's do some "grace period" thingy. E.g. don't clean up MIGs that are younger than X minutes.